pkg/admission/admission.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2021 The Operating System Manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type admissionData struct {
+	client       ctrlruntimeclient.Client
+	ospNamespace string
+}
+
+func New(listenAddress, ospNamespace string, client ctrlruntimeclient.Client) (*http.Server, error) {
+	mux := http.NewServeMux()
+	ad := &admissionData{
+		client:       client,
+		ospNamespace: ospNamespace,
+	}
+
+	mux.HandleFunc("/validate-machine-deployment", handleFuncFactory(ad.validateMachineDeployments))
```

Comment:
usually you only write down the resource that you are validating without type in `validate` since this is an admission webhook and it is not doing anything else but validating :). Can you also use the name of the path similar to the crd name? Like `/machinedeployment`, `operatingsystemprofile`, etc...